### PR TITLE
Story/ccp 117

### PR DESCRIPTION
**Pull request by mkarthik90**
Changes for Editing concepts. Remove and Add and other parameters .

### CODE REVIEW

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/bean/ConceptEditBean.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+	public String getSelectedPosValue() {
+		return selectedPosValue;
+	}
+
+	public void setSelectedPosValue(String selectedPosValue) {
+		this.selectedPosValue = selectedPosValue;
+	}
+
+	public Map<String, String> getPossMap() {
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/bean/ConceptEditBean.java]: spelling mistake? shouldn't it be only one 's'?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/bean/ConceptEditBean.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+		possMap.put("noun", "Nouns");
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/bean/ConceptEditBean.java]: also, check if there isn't an Enum that contains the values and/or labels. It's used in several places, so if there is non, there should be one.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/core/ConceptEntry.java [hunk 3]
```diff
@@ -72,6 +72,16 @@
 
 	private boolean isDeleted;
 
+	public ConceptEntry(){
+
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/core/ConceptEntry.java]: remove empty line and have it all in one line

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/core/ConceptEntry.java [hunk 4]
```diff
@@ -72,6 +72,16 @@
 
 	private boolean isDeleted;
 
+	public ConceptEntry(){
+		
+	}
+	
+	public ConceptEntry(String id,String word,String description){
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/core/ConceptEntry.java]: there should be spaces between paramters (after the comma)

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/core/impl/ConceptManager.java [hunk 5]
```diff
@@ -138,6 +138,18 @@ protected void fillConceptEntry(ConceptEntry entry) {
...
 			entry.setSynonymIds((entry.getSynonymIds() != null ? entry.getSynonymIds() : "")
 					+ Constants.CONCEPT_SEPARATOR + sb.toString());
+			
+			//Since synonymId contains duplicates placing a check to remove duplicates from synonymId
+			String synonymIds = entry.getSynonymIds();
+			String[] synonyms = synonymIds.split(Constants.CONCEPT_SEPARATOR);
+			StringBuilder uniqueSynonym = new StringBuilder("");
+			for(String synonym:synonyms){
+				if(!uniqueSynonym.toString().contains(synonym)){
+				uniqueSynonym.append(synonym);
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/core/impl/ConceptManager.java]: formatting

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/core/impl/ConceptManager.java [hunk 6]
```diff
@@ -223,7 +235,7 @@ protected void fillConceptEntry(ConceptEntry entry) {
 		if (concept == null)
 			return allEntries.toArray(new ConceptEntry[allEntries.size()]);
 
-		fillConceptEntry(concept);
+		//fillConceptEntry(concept);
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/core/impl/ConceptManager.java]: Why did you commented that out?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java [hunk 7]
```diff
@@ -145,26 +113,24 @@ public String cancelEdit(@PathVariable("conceptList") String conceptList) {
 	 *         page
 	 */
 	@RequestMapping(value = "auth/conceptlist/editconcept/edit/{id}", method = RequestMethod.POST)
-	public String confirmlEdit(@PathVariable("id") String id,
-			HttpServletRequest req, Principal principal) {
+	public String confirmlEdit(@PathVariable("id") String id, HttpServletRequest req, Principal principal,
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java]: spelling mistake: confirmlEdit (should be without 'l'?)

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java [hunk 8]
```diff
@@ -180,11 +146,39 @@ public String confirmlEdit(@PathVariable("id") String id,
...
-		return entries;
+	public @ResponseBody ResponseEntity<String> searchConcept(@RequestParam("synonymname") String synonymname) {
+		ConceptEntry[] entries = conceptManager.getConceptListEntriesForWord(synonymname.trim());
+
+		StringBuffer jsonStringBuilder = new StringBuffer("{");
+		jsonStringBuilder.append("\"Total\":");
+		jsonStringBuilder.append(entries.length);
+		jsonStringBuilder.append(",");
+		jsonStringBuilder.append("\"synonyms\":");
+		int i = 0;
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java]: What's that i for? Do you use that?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java [hunk 9]
```diff
@@ -180,11 +146,39 @@ public String confirmlEdit(@PathVariable("id") String id,
 	 * @return The list of existing concepts
 	 */
 	@RequestMapping(method = RequestMethod.GET, value = "conceptEditSynonymView")
-	public @ResponseBody
-	ConceptEntry[] searchConcept(@RequestParam("synonymname") String synonymname) {
-		ConceptEntry[] entries = conceptManager
-				.getConceptListEntriesForWord(synonymname.trim());
-		return entries;
+	public @ResponseBody ResponseEntity<String> searchConcept(@RequestParam("synonymname") String synonymname) {
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java]: I don't think you need the ResponseBody annotation if you return a ResponseEntity. But I'm not sure.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java [hunk 10]
```diff
@@ -180,11 +146,39 @@ public String confirmlEdit(@PathVariable("id") String id,
...
 	@RequestMapping(method = RequestMethod.GET, value = "conceptEditSynonymView")
-	public @ResponseBody
-	ConceptEntry[] searchConcept(@RequestParam("synonymname") String synonymname) {
-		ConceptEntry[] entries = conceptManager
-				.getConceptListEntriesForWord(synonymname.trim());
-		return entries;
+	public @ResponseBody ResponseEntity<String> searchConcept(@RequestParam("synonymname") String synonymname) {
+		ConceptEntry[] entries = conceptManager.getConceptListEntriesForWord(synonymname.trim());
+
+		StringBuffer jsonStringBuilder = new StringBuffer("{");
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java]: Put the generation of Json at least in its own method. If you need it more often, maybe even a helper class.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java [hunk 11]
```diff
@@ -216,13 +213,70 @@ public String confirmlEdit(@PathVariable("id") String id,
 				}
 			}
 		}
-
 		ConceptEntry[] arraySynonyms = new ConceptEntry[synonyms.size()];
+
 		int i = 0;
+		StringBuffer jsonStringBuilder = new StringBuffer("{");
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java]: Here you could use the methods you would create for the above method.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java [hunk 12]
```diff
@@ -216,13 +213,70 @@ public String confirmlEdit(@PathVariable("id") String id,
...
+	 *            A generic model holder for Servlet
+	 * @return synonym details
+	 * @throws JSONException
+	 */
+	@RequestMapping(method = RequestMethod.GET, value = "getConceptAddSynonyms")
+	public @ResponseBody ResponseEntity<String> getSynonymRows(@RequestParam("synonymid") String synonymid,
+			ModelMap model) throws JSONException {
+
+		ConceptEntry synonym = conceptManager.getConceptEntry(synonymid);
+		StringBuffer jsonStringBuilder = new StringBuffer("{");
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java]: and again, create method for json creation and reuse them.

**Comment by jdamerow** [comment]
see comments

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java (hunk 8 above)
Comment (mkarthik90) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java]: @jdamerow  The variable i is used for checking if the synonym we are adding is the last in the synonym list. If that is the last synonym I wont be appending comma to the array in JSON.

**Pull request closed without merging**